### DISCUSSION TITLE
Emulating an eastron-sdm230 - possible?

### DISCUSSION THREAD

**Discussion by That-Dude** [pymodbus-dev/pymodbus#1074] [Q&A] [answered]
Hi

Apologies if this question is out of scope, I just wanted to get some opinions before I start developing.

I would like to emulate an eastron-sdm230 meter that is designed to talk to solar inverters, would this library be suitable for that?

I have a Pi with a USB to RS485 Modbus adaptor and I can read / write registers on the solar inverter.

Now i would like to emulate a meter on the same modbus so that I can feed dummy values to the inverter. There are approximately 24 holding registers to emulate - but I suspect the inverter will only be reading 2 of them.

The registers and values are documented [here](https://xn--stromzhler-v5a.eu/media/pdf/0e/ca/1d/SDM230-Modbus.pdf) in case anyone has a moment to look.

**Comment by janiversen** [comment, marked as answer]
I have no experience with the eastron product line, but it is very likely that the library support it. We support all standard modbus implementation, and if you find a bug we correct it.

There are a few devices out there using a non-standard implementation, for those the library still helps with the communication but you have to program your own decoding/encoding using out custom message facility.

**Reply by That-Dude** [reply]
Thanks for replying!

I'll see if i can work through the code samples and get this up and running.

**Comment by janiversen** [comment]
use client_async.py and client_async_basic_calls.py. If you absolutely want sync code, use client_sync*py
